test/functional/apps/discover/group1/_shared_links.ts

```diff
@@ -87,25 +97,56 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
               pause: true,
             },
           });
+          if (await PageObjects.share.isShareModalOpen()) {
+            await PageObjects.share.closeShareModal();
+          }
         });
 
         it('should allow for copying the snapshot URL as a short URL', async function () {
+          // modal being open blocks test execution
+          if (await PageObjects.share.isShareModalOpen()) {
+            await PageObjects.share.closeShareModal();
+          }
           const re = new RegExp(baseUrl + '/app/r/s/.+$');
-          await PageObjects.share.checkShortenUrl();
+
           await retry.try(async () => {
-            const actualUrl = await PageObjects.share.getSharedUrl();
+            // need to close the modal for the rest of the test to execute on retries
+            if (await PageObjects.share.isShareModalOpen()) {
+              await PageObjects.share.closeShareModal();
+            }
+            if (!(await PageObjects.share.isShareMenuOpen())) {
+              await PageObjects.share.clickShareTopNavButton();
+            }
+            // do not change these testSubjects - this test is very particular
+            await testSubjects.click('Permalinks');
+            await testSubjects.setCheckbox('useShortUrl', 'check');
+            await (
+              await testSubjects.find('createShortUrl')
+            ).waitForDeletedByCssSelector('.euiLoadingSpinner');
+            const actualUrl = await testSubjects.getAttribute(
+              'copyShareUrlButton',
+              'data-share-url'
+            );
             expect(actualUrl).to.match(re);
           });
+          await PageObjects.share.closeShareModal();
         });
 
         it('should allow for copying the saved object URL', async function () {
           const expectedUrl =
             baseUrl + '/app/discover#' + '/view/ab12e3c0-f231-11e6-9486-733b1ac9221a' + '?_g=()';
           await PageObjects.discover.loadSavedSearch('A Saved Search');
           await PageObjects.share.clickShareTopNavButton();
-          await PageObjects.share.exportAsSavedObject();
-          const actualUrl = await PageObjects.share.getSharedUrl();
-          expect(actualUrl).to.be(expectedUrl);
+          await testSubjects.click('Permalinks');
+          await testSubjects.click('exportAsSavedObject');
+          const radioOption = await testSubjects.find('exportAsSavedObject');
+          const [label] = await radioOption.findAllByCssSelector('label[for="savedObject"]');
+          await label.click();
+          await testSubjects.click('copyShareUrlButton');
+          expect(await testSubjects.getAttribute('copyShareUrlButton', 'data-share-url')).to.be(
+            expectedUrl
+          );
+          await PageObjects.share.closeShareModal();
```

Comment:
Same question here as above - why can't we still use `PageObjects.share.exportAsSavedObject()`?

Reply:
This was the same issue with the tests not finding the data-test-subj Permalinks or sharePanel-Permalinks.